### DISCUSSION TITLE
Don't provide ‘’events‘’ in basic components of the system.

### DISCUSSION THREAD

**Discussion by waterflier** [buckyos/buckyos#168] [General]
From the perspective of performance and dependability, **events** in distributed systems are not easy to implement simply. And events are very easy to use. If provided, application developers will definitely use them extensively.

What we should provide is a simple, reliable, and strongly consistent query interface. We can tell the upper layer that the status has changed by carrying the version number in the return.

@alexsunxl

**Comment by alexsunxl** [comment]
OK

**Comment by alexsunxl** [comment]
event-driven architectures provide great flexibility and responsiveness but also introduce complexity, especially when it comes to ensuring performance, reliability, and strong consistency.

Etcd comes with an event mechanism. But I think we can ignore this, and designing a version number in the system configuration service to achieve consistency in system configurations.

Specific features are as follows:
1. State Versioning
2. Simple Query API
   `function getState(key) returns (state, version)`
3. Polling Over Pushing
4. Handling State Changes
5. Optimizing Polling

**Comment by waterflier** [comment]
The Message Bus service (basically Pub/Sub) that we are about to develop will challenge us to continue to face this challenge: Convenient, but not abused
